webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/inventory/InventoryFileWatcher.scala

```diff
@@ -376,59 +413,71 @@ class ProcessOldFiles(
       }
     }
   }
-}
-
-/*
- * A class that periodically add inventories that we likely missed by inotify
- */
-class SchedulerMissedNotify(
-    directories: List[File]
-  , addFiles   : List[File] => UIO[Unit]
-  , period     : Duration
-  , zclock     : Clock
-){
 
   /*
    * List all files that need to be added again - simply all files older than period.
-   * Be careful that files can disapear in the middle of a java.nio.Files.walk, so just
+   * Be careful that files can disappear in the middle of a java.nio.Files.walk, so just
    * don't use that (or better files methods that use that)
    * See: https://issues.rudder.io/issues/19268
    */
   def listFiles(d: Duration): UIO[List[File]] = {
     import scala.jdk.CollectionConverters._
-    (for {
-      // if that fails, just exit
-      ageLimit <- IOResult.effect(DateTime.now().minusMillis(d.toMillis.toInt))
-      filter   =  (f:File) => if(f.exists && InventoryProcessingUtils.hasValidInventoryExtension(f) && (f.isRegularFile && ageLimit.isAfter(f.lastModifiedTime.toEpochMilli))) Some(f) else None
-      // if the listing fails, just exit
-      children <- ZIO.foreach(directories)(d => IOResult.effect(FileUtils.listFilesAndDirs(d.toJava, TrueFileFilter.TRUE, TrueFileFilter.TRUE).asScala))
-      // filter file by file. In case of error, just skip it. Speficically ignore FileNotFound (davfs temp files disapear)
-      filtered <- ZIO.foreach(children.flatten) { file =>
-                    IO.effect(filter(File(file.toPath))).catchAll {
-                      case _:FileNotFoundException => // just ignore
-                        InventoryProcessingLogger.trace(s"Ignoring file '${file.toString}' when processing old inventories: " +
-                                                        s"FileNotFoundException (likely if disapeared between directory listing and filtering)"
-                        ) *> UIO.effectTotal(None)
-                      case ex: Throwable => // log and switch to the next
-                        InventoryProcessingLogger.warn(s"Error when processing file in old inventories: '${file.toString}': ${ex.getMessage}") *>
-                        UIO.effectTotal(None)
-                    }
-                  }
-    } yield (filtered.flatten)).catchAll(err =>
-      InventoryProcessingLogger.warn(s"Error when looking for old inventories that weren't processed: ${err.fullMsg}")*>
+    (
+      for {
+        // if that fails, just exit
+        ageLimit <- IOResult.effect(DateTime.now().minusMillis(d.toMillis.toInt))
+        filter = (f: File) => if (f.exists && InventoryProcessingUtils.hasValidInventoryExtension(f) &&
+                                  (f.isRegularFile && ageLimit.isAfter(f.lastModifiedTime.toEpochMilli)))
+                                Some(f) else None
+
+        // if the listing fails, just exit
+        children <- ZIO.foreach(directories)(d => IOResult.effect(FileUtils.listFilesAndDirs(
+          d.toJava,
```

Comment:
children <-  ZIO.foreach(directories)( d =>
                        IOResult.effect(FileUtils.listFilesAndDirs(d.toJava, TrueFileFilter.TRUE, TrueFileFilter.TRUE  ).asScala)
                     )

would be easier to read (cannot make suggestion on multiple lines ...

Reply:
not perfect indentation though, quite complicated to write through github

Reply:
hum, it seems that intellij changed indentation when it should have not, what a mess

Reply:
that part didn't change, I only move it up

Reply:
Yes , i have seen that it was just moved, and reindent, and yes it clearly feels that it was not rewritten by you !!